lib/assembly/build.ex

```diff
@@ -111,21 +111,27 @@ defmodule Assembly.Build do
   def update_build(build, attrs) do
     with %{changes: changes} = changeset when map_size(changes) > 0 <- Schemas.Build.changeset(build, attrs),
          {:ok, updated_build} <- Repo.update(changeset),
-         preloaded_build <- Repo.preload(updated_build, [:options]) do
-      case Registry.lookup(@registry, to_string(preloaded_build.hal_id)) do
-        [{pid, _value}] ->
-          GenServer.cast(pid, {:update_build, preloaded_build})
-          {:ok, preloaded_build}
-
-        _ ->
-          DynamicSupervisor.start_child(@supervisor, {GenServers.Build, preloaded_build})
-          {:ok, preloaded_build}
-      end
+         preloaded_build <- Repo.preload(updated_build, [:options]),
+         {:ok, _} <- update_preloaded_build(preloaded_build) do
+      {:ok, preloaded_build}
     else
-      # No-op if no changes occure. Avoids sending build updated messages on the
+      # No-op if no changes occur. Avoids sending build updated messages on the
       # queue, recalculating status, etc.
       %{errors: []} -> {:ok, build}
       %{changes: _} = changeset -> {:error, changeset}
+      error -> Logger.error("unexpected error occurred while updating build: #{inspect(error)}")
```

Comment:
Might want to return something like `:error` or a tuple here because Logger will just return `:ok`.

https://hexdocs.pm/logger/1.12.3/Logger.html#error/2

Reply:
thanks, good catch